tests/cpp/multidevice.cpp

```diff
@@ -16,6 +16,14 @@
 
 namespace nvfuser {
 
+void checkSameShardings(TensorView* ref, TensorView* tv) {
```

Comment:
```suggestion
void checkStrictlySameShardings(TensorView* ref, TensorView* tv) {
```
Or smth like that, just to convey that the function is not robust to any axis change/transform (e.g. to broadcast, reduction, transpose, view, split, etc), and to disambiguate from functions like `getTvsWithDifferentSharding`.

Reply:
Looking at the tests, I think it would be actually better to have a more robust function, otherwise the tests will break when they'll become more complex. Why not using `getTvsWithDifferentSharding` directly?

Reply:
> BTW is there a reason getTvsWithDifferentSharding is a templated function?

If I remember correctly this was because it was used with `std::vector` and `std::unordered_set` and I grew tired of modifying it every time :) But I am not sure it's relevant anymore so feel free to simplify it